x-pack/plugins/uptime/public/components/common/charts/__tests__/ping_histogram.test.tsx

```diff
@@ -5,13 +5,42 @@
  */
 
 import React from 'react';
+import DateMath from '@elastic/datemath';
 import { PingHistogramComponent, PingHistogramComponentProps } from '../ping_histogram';
 import { renderWithRouter, shallowWithRouter, MountWithReduxProvider } from '../../../../lib';
-import moment from 'moment';
+
+jest.mock('moment-timezone', () => {
+  return function () {
+    return { tz: { guess: () => 'America/New_York' } };
+  };
+});
+
+jest.mock('moment', () => {
+  return function () {
+    return { fromNow: () => 'a year ago' };
+  };
+});
+
+jest.mock('../../../../../../../../src/plugins/data/public', () => {
+  return function () {
+    return {
+      esKuery: {
+        fromKueryExpression: () => 'an ast',
+        toElasticsearchQuery: () => 'an es query',
+      },
+    };
+  };
+});
```

Comment:
can we move these mocks to a test_util file, so that these can also be used in other places.

Reply:
Do you mean the custom module factories? So use would look like:

```ts
import { momentMockFactory } from './util';
jest.mock('moment', momentMockFactory);
```

We could modularize them, but they're kind of specific to the needs of this test file. `moment-timezone` probably won't need to be mocked in most situations, for example, and there are lots of cases where we'll need more than just `fromNow` on `moment` mocks.

Reply:
can we consolidate all these mocks, so we can use it in other places as well https://github.com/elastic/kibana/blob/master/x-pack/plugins/uptime/public/lib/helper/test_helpers.ts